skillserver/echo.go

```diff
@@ -171,6 +171,11 @@ func (this *EchoResponse) EndSession(flag bool) *EchoResponse {
 	return this
 }
 
+func (this *EchoResponse) Dialog(name string, intent *EchoIntent) *EchoResponse {
```

Comment:
Would be good to document and/or rename this function. From the name, parameters, and return value alone I don't understand what it does.

Reply:
Umm, I was kind of confused. Neither does this method only start a dialog, nor does it end it, its kind of, all states. I can't also say respond becayse this one is a different kind of response. 
Like you have kept .Card func, I thought I could keep .Dialog, can you please suggest a name here ?

Reply:
Happy to suggest a name once I understand what it does :) Can you give me some more details? It looks like it appends a name and an intent to a list of response directives. What does doing that do? Maybe "AddDirective"? That being said why do we need this helper? It's a wrapper around a single line of code.

Reply:
Okay, so the choice of name is from 4 values namely: Dialog.Delegate, Dialog.ElicitSlot, Dialog.ConfirmSlot, Dialog.ConfirmIntent.
All of these 4 have specific purpose when alexa talks on the same intent to the user. Basically, dialog api is for getting "slot" values from the chit chat. Slot values can be stuff like dates, time, singers, authors etc. The response to the opening of intent generally can either be a voice or a dialog response out of the above 4. 
If the user has given all slot values, there is no point opening a dialog, hence this is only set if stuff is missing.
Eg., in a engaging conversation, like questionnaire, alexa will ask a question and expect a response. If alexa would like to hint the user, she would say Dialog.Delegate { updatedSlot: <suggestion> }